spec/Permutations.cry

```diff
@@ -1,25 +1,45 @@
-/**
+/*
  * The functions in this module are derived from the consensus-specs.
  * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#helper-functions
  */
 module Permutations where
 
+import Utils
+
+/**
+ * Check if `value` is a power of two integer.
+ * Return `1:[1]` if true and `0:[1]` otherwise.
+ * NOTE: We use the `Utils::bit_to_bool` function because SAW cannot
+ *  convert the `Bit` type internally to an LLVM integer type.
+ */
+is_power_of_two : {n} (fin n, n >= 1) => [n] -> [1]
+is_power_of_two value = bit_to_bool (is_power_of_two_bit value)
+
 /**
  * Check if `value` is a power of two integer.
  * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#is_power_of_two
  */
-is_power_of_two : {n} (fin n, n >= 1) => [n] -> Bit
-is_power_of_two value = (value > 0) && ((value && (value - 1)) == 0)
+is_power_of_two_bit : {n} (fin n, n >= 1) => [n] -> Bit
+is_power_of_two_bit value = (value > 0) && ((value && (value - 1)) == 0)
 
 /**
  * Reverse the bits of an integer of length `n`.
  * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#reverse_bits
  * NOTE: The Python spec only reverses the low-order bits
  */
-reverse_bits : {n, order_bit_length} (fin n, n >= 1, fin order_bit_length, order_bit_length >= 1, order_bit_length <= n)
+reverse_bits : {n, order_bit_length} (fin n, n >= 1, order_bit_length >= 1, order_bit_length <= n)
```

Comment:
Curious was this necessary for SAW to work or just an unnecessary condition noticed?

Reply:
Just an unnecessary condition since `n` is already noted as finite and `order_bit_length <= n`. So a couple of these commits aren't necessary for getting SAW running, but just clean up commits. Should I note that?